diff --git a/src/wayland/layer_shell_overlay.c b/src/wayland/layer_shell_overlay.c
index bf27c51..3c20885 100644
--- a/src/wayland/layer_shell_overlay.c
+++ b/src/wayland/layer_shell_overlay.c
@@ -32,7 +32,9 @@ int overlay_create(struct wl_display *display, struct wl_compositor *compositor,
 
     ov->surface = wl_compositor_create_surface(compositor);
     ov->layer_surface = zwlr_layer_shell_v1_get_layer_surface(
-        layer_shell, ov->surface, output, ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY, namespace);
+        layer_shell, ov->surface, output,
+        strcmp(namespace, "wayoled-mask") == 0 ? ZWLR_LAYER_SHELL_V1_LAYER_TOP
+                                               : ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY, namespace);
 
     zwlr_layer_surface_v1_add_listener(ov->layer_surface, &layer_surface_listener, ov);
 
@@ -49,6 +51,10 @@ int overlay_create(struct wl_display *display, struct wl_compositor *compositor,
     zwlr_layer_surface_v1_set_exclusive_zone(ov->layer_surface, -1);
     zwlr_layer_surface_v1_set_keyboard_interactivity(ov->layer_surface, 0);
 
+    struct wl_region *input = wl_compositor_create_region(compositor);
+    wl_surface_set_input_region(ov->surface, input);
+    wl_region_destroy(input);
+
     wl_surface_commit(ov->surface);
     wl_display_roundtrip(display);
 
